### PR TITLE
feature/deployment

### PR DESCRIPTION
**Pull request by latusikl**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -54,8 +54,8 @@
               "budgets": [
                 {
                   "type": "initial",
-                  "maximumWarning": "500kb",
-                  "maximumError": "1mb"
+                  "maximumWarning": "1.5mb",
```
Comment (syphzur) [inline angular.json:57]: expensive

File: electron/runner/dev-runner.ts [hunk 2]
```diff
@@ -7,28 +7,47 @@ const ANGULAR_READY = 'angular-ready';
...
 
+function addListenerToElectronProcess(angularChildProcess: ChildProcess, electronChildProcess: ChildProcess) {
+    electronChildProcess.on('exit', () => {
+        angularChildProcess.kill('SIGTERM');
+        process.exit(0);
+    });
+}
+
+function isWindows(): boolean {
+    return process.platform == 'win32';
```
Comment (syphzur) [inline electron/runner/dev-runner.ts:46]: process.platform returns string so use triple-equal operator

**Review by syphzur — CHANGES_REQUESTED** [review]

**Comment by syphzur** [comment]
verified on windows and build-prod script works

File: electron/runner/prod-runner.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import { EventEmitter } from 'events';
+import { Transform, TransformCallback } from 'stream';
+import * as builder from 'electron-builder';
+import * as fs from 'fs';
+
+const ANGULAR_READY = 'angular-ready';
+const DEPLOYMENT_DIR = 'deployment';
+const eventEmitter = new EventEmitter();
+
+function buildForProduction() {
```
Comment (Suchar230) [inline electron/runner/prod-runner.ts:11]: Please add return type :void

File: electron/runner/prod-runner.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            const chunkValue = chunk.toString();
+            if (chunkValue.includes('Build at')) {
+                eventEmitter.emit(ANGULAR_READY);
+            }
+            callback(null, chunkValue);
+        },
+    });
+}
+
+function runElectronBuilder() {
```
Comment (Suchar230) [inline electron/runner/prod-runner.ts:80]: Please add return type

File: tsconfig.json [hunk 5]
```diff
@@ -14,10 +14,9 @@
     "experimentalDecorators": true,
     "moduleResolution": "node",
     "importHelpers": true,
-    "target": "es2015",
+    "target": "es5",
```
Comment (Suchar230) [inline tsconfig.json:17]: is it necessary to lower target version?

Comment (latusikl) [reply tsconfig.json:17]: Yes

**Pull request merged**